### PR TITLE
Drift Detection with Pulumi Deployments Tutorial

### PR DESCRIPTION
**Pull request by interurban**
Adding a new tutorial to :
   Learn to manage infrastructure drift with Pulumi Deployments.

In this tutorial, learn to detect and remediate infrastructure drift in AWS using Pulumi Deployments and GitHub integration.

### CODE REVIEW

File: content/tutorials/drift-detection/index.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
+---
+title_tag: "Manage Infrastructure Drift with Pulumi Deployments | Tutorials"
```
Comment (aaronkao) [inline content/tutorials/drift-detection/index.md:2]: Should the title even mention Pulumi Deployments? It should be more about the use case of what the user wants to accomplish rather than how to use product X.  Remember https://developers.cloudflare.com/style-guide/documentation-content-strategy/content-types/tutorial/

I think that is true with https://www.pulumi.com/tutorials/pulumi-deployments-click-to-deploy/ this might need to get renamed because its not use case focused. Its actually just a how-to for a product. Like the title probably should be "Set up CICD Automation for Infrastructure" or something like that. Then there is probably a tutorial that shows the 3rd party way to do it by wiring it to their existing CICD solution.

File: content/tutorials/drift-detection/index.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+Click **Save deployment configuration**.
+
+![Animation of how to configure drift detection in Pulumi](./pulumi-configure-drift.mp4)
+
+Once you have configured Pulumi Deployments, open the Actions menu and choose `Detect Drift`, then click `Deploy` to validate your stack, program and deployments are all running before we introduce drift.
+
+![Screenshot of the Pulumi Cloud console to run a Drift Detection](./pulumi-detect-drift-action.png)
+
+Once the drift run is complete, you should see a new Drift check on the stack `Drift` Tab
```
Comment (aaronkao) [inline content/tutorials/drift-detection/index.md:156]: What does mean or indicate? You should explain that given there were no deviations  because no drift had been introduced yet.

File: content/tutorials/drift-detection/index.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+Secret environment variables, such as `AWS_SECRET_ACCESS_KEY`, are encrypted end-to-end with Pulumi and can be set on each stack. However, by creating an [environment](/docs/pulumi-cloud/esc) with Pulumi ESC, you can centralize secrets and set up OIDC for secure authentication. This allows you to manage and share sensitive configuration data across multiple stacks efficiently.
+{{% /notes %}}
+
+Click **Save deployment configuration**.
+
+![Animation of how to configure drift detection in Pulumi](./pulumi-configure-drift.mp4)
+
+Once you have configured Pulumi Deployments, open the Actions menu and choose `Detect Drift`, then click `Deploy` to validate your stack, program and deployments are all running before we introduce drift.
+
+![Screenshot of the Pulumi Cloud console to run a Drift Detection](./pulumi-detect-drift-action.png)
```
Comment (aaronkao) [inline content/tutorials/drift-detection/index.md:154]: this is a screenshot of a failed detection which is not what they will see if there is no drift yet

File: content/tutorials/drift-detection/index.md [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+
+Once the drift run is complete, you should see a new Drift check on the stack `Drift` Tab
+
+## Introduce drift
+
+To see Drift Detection in action, you will now introduce drift by manually making changes to your newly created EC2 resource, in this case by adding a new tag.
+
+Log in to the AWS Management console with your browser and ensure you are in the correct AWS region where your Pulumi stack is deployed.
+
+Modify your EC2 Instance by going to the EC2 Dashboard, then select the EC2 instance named `webserver` and click on the **Tags** tab. Next, add a tag by clicking the **Manage tags** button. Then click the **Add new tag** button with the key `Description` and value of `Pulumi Drift Detection`. Finally, **Save** the changes.
```
Comment (aaronkao) [inline content/tutorials/drift-detection/index.md:164]: i would maybe do a tab chooser so you can give the Management console version as well as the AWS CLI version. The CLI version is faster and doesn't require you to leave the command line

File: content/tutorials/drift-detection/index.md [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+
+## Run a Drift Detection deployment
```
Comment (aaronkao) [inline content/tutorials/drift-detection/index.md:166]: Is that really the correct way to say it? Should it isntead be "Run Drift Detection" or "Detect drift" or something that is more about the goal of the action rather than the feature of the product?

File: content/tutorials/drift-detection/index.md [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+
+After your run completes you will see a warning icon on the Drift tab, indicating that drift has been detected. You can see a detailed summary of what resources have been updated or deleted, the properties that have changed, when the drift run happened and a link to the Deployment with further details.
+
+![Animation of the Pulumi Cloud console Drift tab](./pulumi-drift-detected.mp4)
+
+## Remediate Drift
+
+Now that you have detected and reviewed your infrastructure drift, you can choose to remediate and overwrite the changes made in your cloud provider with the most recently specified desired state of your Pulumi program. This can be done via the **Remediate drift** option in the **Actions** menu on your stack page.
+
+Alternatively, you can accept the changes made in your cloud provider and apply them to your Pulumi state. This can be done via the **Refresh** option in the **Actions** drop down on your stack page. You will also want to update your Pulumi program to align with the changes made in the cloud provider.
```
Comment (aaronkao) [inline content/tutorials/drift-detection/index.md:178]: Is there no option for having it auto update your Pulumi program for you or generating the right code to replace it?

File: content/tutorials/drift-detection/index.md [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+
+As the final step of this tutorial select the **Remediate drift** option in the Actions drop down on your stack page to remediate the manual changes. Once the deployment is complete, your infrastructure will be restored to the desired state expressed in your code.
```
Comment (aaronkao) [inline content/tutorials/drift-detection/index.md:180]: I do think this is missing something which is talking about how you can create drift detection on a schedule. Its a very easy mention that doesn't cost much time or space

**Review by aaronkao — CHANGES_REQUESTED** [review]
This is a good start. I think there are a few things missing. Let's get those in. Thanks

File: content/tutorials/drift-detection/index.md [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+
+Modify your EC2 Instance by going to the EC2 Dashboard, then select the EC2 instance named `webserver` and click on the **Tags** tab. Next, add a tag by clicking the **Manage tags** button. Then click the **Add new tag** button with the key `Description` and value of `Pulumi Drift Detection`. Finally, **Save** the changes.
+
+## Run a Drift Detection deployment
+
+Return to the Pulumi Cloud and your stack, and again choose the **Detect Drift** action and click **Deploy**.
+
+After your run completes you will see a warning icon on the Drift tab, indicating that drift has been detected. You can see a detailed summary of what resources have been updated or deleted, the properties that have changed, when the drift run happened and a link to the Deployment with further details.
+
+![Animation of the Pulumi Cloud console Drift tab](./pulumi-drift-detected.mp4)
```
Comment (sean1588) [inline content/tutorials/drift-detection/index.md:172]: for some reason none of these mp4 files are rendering for me.

File: content/tutorials/drift-detection/index.md (hunk 6 above)
Comment (sean1588) [reply content/tutorials/drift-detection/index.md:178]: I don't think there is for refresh as far as I know, unless this was something recently added. It only updates the state to true it up with the current state of the deployed resources, but does not give the code to update the resource's configuration in the program.

File: content/tutorials/drift-detection/index.md (hunk 5 above)
Comment (sean1588) [reply content/tutorials/drift-detection/index.md:166]: agreed here.

Comment (sean1588) [reply content/tutorials/drift-detection/index.md:166]: agreed here

File: content/tutorials/drift-detection/index.md (hunk 6 above)
Comment (sean1588) [reply content/tutorials/drift-detection/index.md:178]: I don't think there is for refresh as far as I know, unless this was something recently added. It only updates the state to true it up with the current state of the deployed resources, but does not give the code to update the resource's configuration in the program.

File: content/tutorials/drift-detection/index.md (hunk 2 above)
Comment (interurban) [reply content/tutorials/drift-detection/index.md:156]: thx; added more detail for clarity.

File: content/tutorials/drift-detection/index.md (hunk 3 above)
Comment (interurban) [reply content/tutorials/drift-detection/index.md:154]: fixing screenshot

File: content/tutorials/drift-detection/index.md (hunk 4 above)
Comment (interurban) [reply content/tutorials/drift-detection/index.md:164]: Added the CLI section, thx

File: content/tutorials/drift-detection/index.md (hunk 5 above)
Comment (interurban) [reply content/tutorials/drift-detection/index.md:166]: thx; fixed up

File: content/tutorials/drift-detection/index.md (hunk 6 above)
Comment (interurban) [reply content/tutorials/drift-detection/index.md:178]: That's correct, it does not give you the right code. Just shows you what's changed and you can either choose to overwrite the changes, or accept them and update the state.

File: content/tutorials/drift-detection/index.md (hunk 8 above)
Comment (interurban) [reply content/tutorials/drift-detection/index.md:172]: fixed it; had to convert since I'm using mp4s now instead of animated gifs thx for the catch!

File: content/tutorials/drift-detection/index.md (hunk 7 above)
Comment (interurban) [reply content/tutorials/drift-detection/index.md:180]: Added a section on this, but also plan to cover this in the next drift tutorial on Automation and notification, where i'll map out all the steps.

File: content/tutorials/drift-detection/index.md (hunk 1 above)
Comment (interurban) [reply content/tutorials/drift-detection/index.md]: thx Aaron; I've adjusted the title.

File: content/tutorials/drift-detection-and-remediation/index.md [hunk 9]
```diff
@@ -0,0 +1,224 @@
...
+
+After the drift detection is complete, you will see a new Drift run on the `Drift` tab timeline. Since this is the initial check and no deviations have been introduced, you will not receive any notifications indicating that drift has been detected, nor will you see any resources with changes.
+
+## Introduce drift
+
+To see Drift Detection in action, you will now introduce drift by manually making changes to your newly created EC2 resource, in this case by adding a new tag.
+
+### Add a tag to EC2 using AWS CLI
+
+To add a tag via the AWS CLI you will first need to retrieve the EC2 Instance ID before you can add a tag. Use the the following command:
```
Comment (sean1588) [inline content/tutorials/drift-detection-and-remediation/index.md:167]: _Just my opinion here:_

Can we add a sentence here to explicitly call out this can be done one of two ways, like "This can be done using the AWS CLI or using the AWS console"? It's probably already somewhat obvious, but If reading this tutorial top to bottom as is, you would first run into adding the tag via cli and then move to the next section which is adding via the console which looks like another step to complete. This is just so it is clear this is an either or and the user doesn't do the same step twice.

File: content/tutorials/drift-detection-and-remediation/index.md [hunk 10]
```diff
@@ -0,0 +1,224 @@
...
+
+![Screenshot of the Pulumi Cloud console to run a Drift Detection](./pulumi-detect-drift-action.png)
+
+After the drift detection is complete, you will see a new Drift run on the `Drift` tab timeline. Since this is the initial check and no deviations have been introduced, you will not receive any notifications indicating that drift has been detected, nor will you see any resources with changes.
+
+## Introduce drift
+
+To see Drift Detection in action, you will now introduce drift by manually making changes to your newly created EC2 resource, in this case by adding a new tag.
+
+### Add a tag to EC2 using AWS CLI
```
Comment (sean1588) [inline content/tutorials/drift-detection-and-remediation/index.md:165]: Might be a little verbose, but sounds more accurate imo as EC2 is the general name for the aws service which encompasses a bunch of other resources that fall under it, an EC2 instance being one of them.

```suggestion
### Add a tag to the EC2 instance using AWS CLI
```

File: content/tutorials/drift-detection-and-remediation/index.md [hunk 11]
```diff
@@ -0,0 +1,224 @@
...
+
+Now, add the tag with the key `Description` and the value `Pulumi Drift Detection`:
+
+```bash
+   aws ec2 create-tags --resources <instance-id> --tags Key=Description,Value="Pulumi Drift Detection"
+```
+
+Replace `<instance-id>` with the actual instance ID of your EC2 instance.
+
+### Add a tag to EC2 using the AWS console
```
Comment (sean1588) [inline content/tutorials/drift-detection-and-remediation/index.md:181]: same as  above

```suggestion
### Add a tag to the EC2 instance using the AWS console
```

File: content/tutorials/drift-detection-and-remediation/index.md [hunk 12]
```diff
@@ -0,0 +1,224 @@
...
+   - The [Pulumi CLI](/docs/install/)
+   - A [Pulumi Cloud account](https://app.pulumi.com/signup) and organization subscribed to the Enterprise or Business Critical editions
+   - An [Amazon Web Services](https://aws.amazon.com/) account, access key ID, and secret access key
+   - The [AWS CLI](https://aws.amazon.com/cli/)
+   - Git installed and a [GitHub account](https://github.com/) with admin rights to a repository or organization
+estimated_time: 15
+---
+
+[Drift Detection and Remediation](/docs/pulumi-cloud/deployments/drift/) is a feature of Pulumi Cloud that automates the detection and correction of drift in your cloud environments. It integrates seamlessly with Pulumi Deployments and can be incorporated into CI/CD workflows and tools using the [Pulumi CLI](/docs/cli/) or [Automation API](docs/using-pulumi/automation-api/). When changes occur outside of Pulumi, such as manual updates in the cloud provider's console, Pulumi reports these discrepancies. Auto-remediation can also be configured to refresh your last known desired state, correcting any drift without manual intervention.
```
Comment (aaronkao) [inline content/tutorials/drift-detection-and-remediation/index.md:26]: I think you should mention that you can run drift detection inside your own CICD system as well. It doesn't need to only be configured with Deployments. You should link to that section of the docs if people want to do that.

File: content/tutorials/drift-detection-and-remediation/index.md [hunk 13]
```diff
@@ -0,0 +1,224 @@
...
+
+## Configure Drift Detection
+
+Now you will return to your browser to set your deployment configuration source control settings. Navigate to your stack `Settings` and `Deploy` tab. Next, configure the required Deployment settings, including your source control.
+
+1. Add your GitHub organization/repository and the branch you created.
+
+2. Add your Secret environment variables, including `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`.
+
+{{% notes type="info" %}}
```
Comment (aaronkao) [inline content/tutorials/drift-detection-and-remediation/index.md:144]: yay well done. crosslinking and promoting ESC is good!

File: content/tutorials/drift-detection-and-remediation/index.md (hunk 11 above)
Comment (aaronkao) [inline content/tutorials/drift-detection-and-remediation/index.md:181]: cant you shove this stuff into a tabbed chooser module? otherwise you need to make it clear these are the same instructions executed 2 different ways..

File: content/tutorials/drift-detection-and-remediation/index.md [hunk 14]
```diff
@@ -0,0 +1,224 @@
...
+
+Once you have configured Pulumi Deployments, open the Actions menu and choose `Detect Drift`, then click `Deploy` to validate your stack, program and deployments are all running before we introduce drift.
+
+![Screenshot of the Pulumi Cloud console to run a Drift Detection](./pulumi-detect-drift-action.png)
+
+After the drift detection is complete, you will see a new Drift run on the `Drift` tab timeline. Since this is the initial check and no deviations have been introduced, you will not receive any notifications indicating that drift has been detected, nor will you see any resources with changes.
+
+## Introduce drift
+
+To see Drift Detection in action, you will now introduce drift by manually making changes to your newly created EC2 resource, in this case by adding a new tag.
```
Comment (aaronkao) [inline content/tutorials/drift-detection-and-remediation/index.md:163]: I think you should make a suggestion that they can change anything else they want about the created resources as well to introduce drift.

File: content/tutorials/drift-detection-and-remediation/index.md [hunk 15]
```diff
@@ -0,0 +1,224 @@
+---
+title_tag: "Detect and Remediate Infrastructure Drift"
```
Comment (aaronkao) [inline content/tutorials/drift-detection-and-remediation/index.md:2]: Why don't we push through changes to the deployments tutorials title as part of this PR. Yeah I like what you did to the title on this one. Its much more use cases focused, thanks for doing it

**Review by aaronkao — COMMENTED** [review]
Here is my review. After these changes are made, you are good to go to publish without needing anything else from me. I am going to let someone else approve to make sure its technically correct and good

File: content/tutorials/drift-detection-and-remediation/index.md [hunk 16]
```diff
@@ -0,0 +1,224 @@
...
+To avoid incurring any unwanted charges, clean up the resources you created:
+
+1. Navigate to your stack in Pulumi Cloud and open the **Actions** menu.
+2. Select the **Destroy** option to delete all resources in this stack, and click the confirmation checkbox.
+3. Delete the stack by navigating to the **Settings** tab and choose **Delete stack** to remove the stack, including all its deployment history and state files from Pulumi Cloud.
+4. Delete the GitHub repository you created.
+
+## Next Steps
+
+In this tutorial, you created an EC2 instance with Pulumi, including a VPC with a public subnet, an internet gateway, and a security group allowing HTTP traffic. Then, you introduced drift by manually adding a tag to your EC2 resource via the AWS console. Finally, you learned how to detect your drifted infrastructure and how to remediate the manual changes.
```
Comment (cnunciato) [inline content/tutorials/drift-detection-and-remediation/index.md:219]: I'd definitely reword this to put the important stuff first -- that what you did in this tutorial was learn how to set up automated drift detection and remediation with Pulumi Deployments. (That you happened to use an EC2 instance with a VPC, subnet, gateway, etc., is ultimately irrelevant.)

File: content/tutorials/drift-detection-and-remediation/index.md [hunk 17]
```diff
@@ -0,0 +1,224 @@
...
+
+To learn more about Drift Detection with Pulumi, take a look at the following resources:
+
+- Learn more about [Pulumi Drift Detection](docs/pulumi-cloud/deployments/drift/).
+- Learn more about [Pulumi Deployments](/docs/pulumi-cloud/deployments/).
```
Comment (cnunciato) [inline content/tutorials/drift-detection-and-remediation/index.md:224]: I'd probably mention learning more about Pulumi refresh here, since that's what drift detection is based on, and if we choose not to cover setting up the schedule here (which I think we should!), we should mention how to set that up also here.

**Review by cnunciato — CHANGES_REQUESTED** [review]

File: content/tutorials/drift-detection-and-remediation/index.md (hunk 11 above)
Comment (cnunciato) [reply content/tutorials/drift-detection-and-remediation/index.md:181]: Agree -- we don't need both of these. Suggest just using the console version. It's simpler.

Comment (aaronkao) [reply content/tutorials/drift-detection-and-remediation/index.md:181]: I think I requested in the last edit to add a CLI version since its actually faster to do that since you already have the credentials in the CLI

File: content/tutorials/drift-detection-and-remediation/index.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (cnunciato) [reply content/tutorials/drift-detection-and-remediation/index.md:26]: Started a doc on this so we can get it all into one place.

File: content/tutorials/drift-detection-and-remediation/index.md (hunk 9 above)
Comment (interurban) [reply content/tutorials/drift-detection-and-remediation/index.md:167]: thx, I'll go with the console version

File: content/tutorials/drift-detection-and-remediation/index.md (hunk 11 above)
Comment (interurban) [reply content/tutorials/drift-detection-and-remediation/index.md:181]: thx added

File: content/tutorials/drift-detection-and-remediation/index.md (hunk 12 above)
Comment (interurban) [reply content/tutorials/drift-detection-and-remediation/index.md:26]: thx, feedback incorporated.

File: content/tutorials/drift-detection-and-remediation/index.md (hunk 10 above)
Comment (interurban) [reply content/tutorials/drift-detection-and-remediation/index.md:165]: done, thx

File: content/tutorials/drift-detection-and-remediation/index.md (hunk 11 above)
Comment (interurban) [reply content/tutorials/drift-detection-and-remediation/index.md:181]: dropped the console version

File: content/tutorials/drift-detection-and-remediation/index.md (hunk 15 above)
Comment (interurban) [reply content/tutorials/drift-detection-and-remediation/index.md:2]: going to add this as part of a new PR,, this one's now way behind main

File: content/tutorials/drift-detection-and-remediation/index.md [hunk 18]
```diff
@@ -0,0 +1,224 @@
+---
+title_tag: "Detect and Remediate Infrastructure Drift"
+title: "Detect and Remediate Infrastructure Drift"
+layout: single
+description: |
+   Learn to detect and remediate infrastructure drift with Pulumi Deployments.
+meta_desc: In this tutorial, learn to detect and remediate infrastructure drift in AWS using Pulumi Deployments and GitHub integration.
+meta_image: meta.png
+weight: 71
```
[the remark this answers is not in the document]
Comment (interurban) [reply content/tutorials/drift-detection-and-remediation/index.md:9]: pushing it down, yes we need to re-order all now that we've had a big push of new content

File: content/tutorials/drift-detection-and-remediation/index.md [hunk 19]
```diff
@@ -0,0 +1,224 @@
...
+
+Drift detection can be initiated via the CLI, performed on-demand in Pulumi Cloud with [Click to Deploy](/docs/pulumi-cloud/deployments/#deployment-triggers) actions, or [scheduled](/docs/pulumi-cloud/deployments/schedules/) to run at regular intervals. During each run, Pulumi compares the actual state of your cloud resources with the expected state stored in Pulumi Cloud. Any discrepancies are logged in the Drift tab, which also provides a history of detection runs, details of the drift detected, and triggered notifications. [Notifications](docs/pulumi-cloud/deployments/drift/#configuring-notifications-for-drift-detection) can be sent via webhooks, Slack, or Microsoft Teams and detail the nature and scope of the drift.
```
[the remark this answers is not in the document]
Comment (interurban) [reply content/tutorials/drift-detection-and-remediation/index.md:28]: thx adjusted

File: content/tutorials/drift-detection-and-remediation/index.md (hunk 14 above)
Comment (interurban) [reply content/tutorials/drift-detection-and-remediation/index.md:163]: adjusted the intro

File: content/tutorials/drift-detection-and-remediation/index.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (interurban) [reply content/tutorials/drift-detection-and-remediation/index.md:26]: fixed capitalizations for consistency

File: content/tutorials/drift-detection-and-remediation/index.md [hunk 20]
```diff
@@ -0,0 +1,224 @@
...
+
+To get started with Drift Detection in Pulumi, we will use Pulumi Deployments, which runs on Pulumi-hosted compute.
```
[the remark this answers is not in the document]
Comment (interurban) [reply content/tutorials/drift-detection-and-remediation/index.md:30]: fixed

File: content/tutorials/drift-detection-and-remediation/index.md [hunk 21]
```diff
@@ -0,0 +1,224 @@
...
+you'll_learn:
+ - How to set up and configure Pulumi Deployments for drift detection
+ - How to introduce and identify infrastructure drift in an AWS environment
+ - How to remediate drift ensuring your cloud infrastructure matches your desired state
+prereqs:
+   - Completion of the [Getting Started](/docs/get-started/) guide or familiarity with the basics of the Pulumi workflow
+   - The [Pulumi CLI](/docs/install/)
+   - A [Pulumi Cloud account](https://app.pulumi.com/signup) and organization subscribed to the Enterprise or Business Critical editions
+   - An [Amazon Web Services](https://aws.amazon.com/) account, access key ID, and secret access key
+   - The [AWS CLI](https://aws.amazon.com/cli/)
```
[the remark this answers is not in the document]
Comment (interurban) [reply content/tutorials/drift-detection-and-remediation/index.md:21]: added

File: content/tutorials/drift-detection-and-remediation/index.md (hunk 16 above)
Comment (interurban) [reply content/tutorials/drift-detection-and-remediation/index.md:219]: done

File: content/tutorials/drift-detection-and-remediation/index.md (hunk 17 above)
Comment (interurban) [reply content/tutorials/drift-detection-and-remediation/index.md:224]: done

File: content/tutorials/drift-detection-and-remediation/index.md [hunk 22]
```diff
@@ -0,0 +1,224 @@
...
+git branch -M main
+git remote add origin https://github.com/your-username/learn-pulumi-drift-detection.git
+git push -u origin main
+```
+
+## Configure Drift Detection
+
+Now you will return to your browser to set your deployment configuration source control settings. Navigate to your stack `Settings` and `Deploy` tab. Next, configure the required Deployment settings, including your source control.
+
+1. Add your GitHub organization/repository and the branch you created.
```
[the remark this answers is not in the document]
Comment (interurban) [reply content/tutorials/drift-detection-and-remediation/index.md:140]: added

File: content/tutorials/drift-detection-and-remediation/index.md [hunk 23]
```diff
@@ -0,0 +1,219 @@
...
+
+## Next steps
+
+In this tutorial, you learned how to set up automated drift detection and remediation with Pulumi Deployments. First you created an EC2 instance, including a VPC with a public subnet, an internet gateway, and a security group allowing HTTP traffic. Then, you introduced drift by manually adding a tag to your EC2 resource via the AWS console.
+
+To learn more about drift detection with Pulumi, take a look at the following resources:
+
+- Learn more about [Pulumi drift detection](docs/pulumi-cloud/deployments/drift/).
+- Learn more about [Pulumi Deployments](/docs/pulumi-cloud/deployments/).
+- Learn more about [Pulumi refresh](/docs/iac/cli/commands/pulumi_refresh/)
```
Comment (cnunciato) [inline content/tutorials/drift-detection-and-remediation/index.md:219]:
```suggestion
- [Pulumi drift detection](/docs/pulumi-cloud/deployments/drift/)
- [Pulumi Deployments](/docs/pulumi-cloud/deployments/)
- [Pulumi refresh](/docs/iac/cli/commands/pulumi_refresh/)
```

File: content/tutorials/drift-detection-and-remediation/index.md [hunk 24]
```diff
@@ -0,0 +1,219 @@
...
+
+Follow the prompts to set up your project and accept the default values or specify new values, such as `aws-region`.
+
+Run `pulumi preview` to see the changes that will be made.
+
+```bash
+pulumi preview
+```
+
+Now you can deploy the stack to see the outputs of your EC2 instance, in this case `hostname`, `ip` and `url` which you will use to introduce manual changes via the AWS console.
```
Comment (cnunciato) [inline content/tutorials/drift-detection-and-remediation/index.md:89]:
```suggestion
Now you can deploy the stack to see the outputs of your EC2 instance, which in this case will be `hostname`, `ip` and `url`.
```

**Review by cnunciato — APPROVED** [review]
